internal/controller/maintenancemode_controller.go

```diff
@@ -0,0 +1,199 @@
+package controller
+
+import (
+	"context"
+	"fmt"
+	"sigs.k8s.io/controller-runtime/pkg/event"
+	"time"
+
+	"github.com/red-hat-storage/ocs-client-operator/api/v1alpha1"
+
+	"github.com/go-logr/logr"
+	ramenv1alpha1 "github.com/ramendr/ramen/api/v1alpha1"
+	providerclient "github.com/red-hat-storage/ocs-operator/services/provider/api/v4/client"
+	"k8s.io/apimachinery/pkg/api/meta"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	"k8s.io/apimachinery/pkg/runtime"
+	ctrl "sigs.k8s.io/controller-runtime"
+	"sigs.k8s.io/controller-runtime/pkg/builder"
+	"sigs.k8s.io/controller-runtime/pkg/client"
+	"sigs.k8s.io/controller-runtime/pkg/handler"
+	"sigs.k8s.io/controller-runtime/pkg/log"
+	"sigs.k8s.io/controller-runtime/pkg/predicate"
+	"sigs.k8s.io/controller-runtime/pkg/reconcile"
+)
+
+const (
+	MaintenanceModeCRDName = "maintenancemodes.ramendr.openshift.io"
+)
+
+// MaintenanceModeReconciler reconciles a ClusterVersion object
+type MaintenanceModeReconciler struct {
+	client.Client
+	Scheme *runtime.Scheme
+
+	log logr.Logger
+	ctx context.Context
+}
+
+// SetupWithManager sets up the controller with the Manager.
+func (r *MaintenanceModeReconciler) SetupWithManager(mgr ctrl.Manager) error {
+	generationChangePredicate := predicate.GenerationChangedPredicate{}
+	maintenanceModeChangedPredicate := predicate.Funcs{
+		CreateFunc: func(_ event.CreateEvent) bool {
+			return false
+		},
```

Comment:
You cannot ignore create events, as you cannot assume that the client staus will start with an MM set to false